### PR TITLE
test: Migrate tests to github.com/go-quicktest/qt

### PR DESCRIPTION
**Pull request by sayboras**
Fixes: #1251

### CODE REVIEW

**Comment by lmb** [comment]
@sayboras let me know when you've had a chance to rebase.

**Comment by sayboras** [comment]
> let me know when you've had a chance to rebase.

sure, I didn't notice that my local fork was using master instead of main.

File: go.sum
Comment (lmb) [inline go.sum:1]: Did you run `go mod tidy`?

File: info_test.go [hunk 1]
```diff
@@ -122,14 +123,14 @@ func TestProgramInfo(t *testing.T) {
...
 			} else {
 				uid, ok := info.CreatedByUID()
 				if testutils.IsKernelLessThan(t, "4.15") {
-					qt.Assert(t, ok, qt.IsFalse)
+					qt.Assert(t, qt.IsFalse(ok))
 				} else {
-					qt.Assert(t, ok, qt.IsTrue)
-					qt.Assert(t, uid, qt.Equals, uint32(os.Getuid()))
+					qt.Assert(t, qt.IsTrue(ok))
+					qt.Assert(t, qt.Equals(uid, uint32(os.Getuid())))
```
Comment (lmb) [inline info_test.go:133]: Nit: is the `uint32` conversion still needed?

File: link/uprobe_test.go [hunk 2]
```diff
@@ -54,52 +52,48 @@ func TestExecutableOffset(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	c.Assert(offset, qt.Equals, uint64(0x1))
+	qt.Assert(t, qt.Equals(offset, uint64(0x1)))
```
Comment (lmb) [inline link/uprobe_test.go:55]: Nit: unnecessary type cast?

File: linker_test.go [hunk 3]
```diff
@@ -112,34 +112,33 @@ func TestForwardFunctionDeclaration(t *testing.T) {
...
 	insns := asm.Instructions{
 		asm.Return().WithSymbol("sym"),
 	}
 	m, err := splitSymbols(insns)
-	c.Assert(err, qt.IsNil, qt.Commentf("insns with a single Symbol"))
+	qt.Assert(t, qt.IsNil(err), qt.Commentf("insns with a single Symbol"))
 
-	c.Assert(len(m), qt.Equals, 1)
-	c.Assert(len(m["sym"]), qt.Equals, 1)
+	qt.Assert(t, qt.Equals(len(m), 1))
```
Comment (lmb) [inline linker_test.go:133]: HasLen?

File: linker_test.go [hunk 4]
```diff
@@ -157,11 +156,11 @@ func TestSplitSymbols(t *testing.T) {
...
 	})
-	c.Assert(err, qt.IsNil, qt.Commentf("insns with multiple Symbols"))
+	qt.Assert(t, qt.IsNil(err), qt.Commentf("insns with multiple Symbols"))
 
-	c.Assert(len(m), qt.Equals, 4)
-	c.Assert(len(m["sym1"]), qt.Equals, 1)
-	c.Assert(len(m["sym2"]), qt.Equals, 2)
-	c.Assert(len(m["sym3"]), qt.Equals, 3)
-	c.Assert(len(m["sym4"]), qt.Equals, 4)
+	qt.Assert(t, qt.Equals(len(m), 4))
```
Comment (lmb) [inline linker_test.go:161]: HasLen again?

File: perf/reader_test.go [hunk 5]
```diff
@@ -165,15 +165,15 @@ func checkRecord(tb testing.TB, rd *Reader) (id int, remaining int) {
...
-	qt.Assert(tb, rec.CPU >= 0, qt.IsTrue, qt.Commentf("Record has invalid CPU number"))
+	qt.Assert(tb, qt.IsTrue(rec.CPU >= 0), qt.Commentf("Record has invalid CPU number"))
 
 	size := int(rec.RawSample[0])
-	qt.Assert(tb, len(rec.RawSample) >= size, qt.IsTrue, qt.Commentf("RawSample is at least size bytes"))
+	qt.Assert(tb, qt.IsTrue(len(rec.RawSample) >= size), qt.Commentf("RawSample is at least size bytes"))
 
 	for i, v := range rec.RawSample[2:size] {
-		qt.Assert(tb, v, qt.Equals, byte(0xff), qt.Commentf("filler at position %d should match", i+2))
+		qt.Assert(tb, qt.Equals(v, byte(0xff)), qt.Commentf("filler at position %d should match", i+2))
```
Comment (lmb) [inline perf/reader_test.go:176]: Drop `byte`?

File: perf/reader_test.go [hunk 6]
```diff
@@ -481,12 +481,12 @@ func TestPause(t *testing.T) {
 
 	// Pause/Resume after close should be no-op.
 	err = rd.Pause()
-	qt.Assert(t, err, qt.Not(qt.Equals), ErrClosed, qt.Commentf("returns unwrapped ErrClosed"))
-	qt.Assert(t, errors.Is(err, ErrClosed), qt.IsTrue, qt.Commentf("doesn't wrap ErrClosed"))
+	qt.Assert(t, qt.Not(qt.Equals(err, ErrClosed)), qt.Commentf("returns unwrapped ErrClosed"))
+	qt.Assert(t, qt.IsTrue(errors.Is(err, ErrClosed)), qt.Commentf("doesn't wrap ErrClosed"))
```
Comment (lmb) [inline perf/reader_test.go:485]: `qt.ErrorIs`

File: perf/reader_test.go [hunk 7]
```diff
@@ -481,12 +481,12 @@ func TestPause(t *testing.T) {
...
 
 	err = rd.Resume()
-	qt.Assert(t, err, qt.Not(qt.Equals), ErrClosed, qt.Commentf("returns unwrapped ErrClosed"))
-	qt.Assert(t, errors.Is(err, ErrClosed), qt.IsTrue, qt.Commentf("doesn't wrap ErrClosed"))
+	qt.Assert(t, qt.Not(qt.Equals(err, ErrClosed)), qt.Commentf("returns unwrapped ErrClosed"))
+	qt.Assert(t, qt.IsTrue(errors.Is(err, ErrClosed)), qt.Commentf("doesn't wrap ErrClosed"))
```
Comment (lmb) [inline perf/reader_test.go:489]: Ditto

**Review by lmb — CHANGES_REQUESTED** [review]
Phew! Looks pretty good to me, there is one change that isn't correct I think. A couple of nits with drive by cleanups if you feel like it.

File: info_test.go (hunk 1 above)
Comment (sayboras) [reply info_test.go:133]: yes, here is the signature of Equal function `Equals[T any](got, want T) Checker`, so got and want should have a same type.

File: linker_test.go (hunk 3 above)
Comment (sayboras) [reply linker_test.go:133]: good point :heavy_check_mark:

File: cmd/bpf2go/output_test.go [hunk 8]
```diff
@@ -57,16 +57,12 @@ func TestOrderTypes(t *testing.T) {
...
 			result, err := sortTypes(test.in)
-			qt.Assert(t, err, qt.IsNotNil)
-			qt.Assert(t, result, qt.IsNil)
+			qt.Assert(t, qt.IsNotNil(err))
+			qt.Assert(t, qt.IsNil(result))
 		})
 	}
 }
 
-var typesEqual = qt.CmpEquals(cmp.Comparer(func(a, b btf.Type) bool {
```
[the remark this answers is not in the document]
Comment (sayboras) [reply cmd/bpf2go/output_test.go:66]: good spot :+1:

**Review by lmb — APPROVED** [review]

**Comment by lmb** [comment]
Thanks!

**Pull request merged**